File: Assignment_43/Program2.c
/* Write a recursive program which accept number from user and return largest digit.
Input : 87983
Output : 9
*/

#include <stdio.h>

int Max(int iNo)
{
   static int iMax = 0;
   int iDigit = 0;

   if(iNo < 0)
   {
      iNo = -iNo;
   }

   if(iNo != 0)
   {
        iDigit = iNo % 10;
        if(iDigit > iMax)
        {
            iMax = iDigit;
        }
        iNo = iNo / 10;

        Max(iNo);
   }
        return iMax;
}

int main()
{
    int iValue = 0, iRet = 0;

    printf("Enter number :");
    scanf("%d",&iValue);

    iRet = Max(iValue);

    printf("Largest Digit is : %d\n",iRet);

    return 0;
}

File: Assignment_44/Program2.java
//Write a program which accept matrix and one number from user and return frequency of that number.

import java.util.*;

class Matrix
{
    private int Arr[][];

    public Matrix(int A, int B)
    {
        Arr = new int [A][B];
    }

    public void Accept()
    {
        System.out.println("Please enter the elements of matrix : ");

        Scanner sobj = new Scanner(System.in);

        int i = 0, j = 0;

        for(i = 0; i < Arr.length; i++)
        {
            for(j = 0; j < Arr[i].length; j++)
            {
                Arr[i][j] = sobj.nextInt();
             }
            System.out.println();
        }
    }

     public void Display()
    {
        System.out.println("Elements of the matrix are : ");

        int i = 0, j = 0;

        for(i = 0; i < Arr.length; i++)
        {
            for(j = 0; j < Arr[i].length; j++)
            {
                System.out.print(Arr[i][j]+"\t");
            }
            System.out.println();
        }
    }

    public int Frequency(int iNo)
    {
       int iCount = 0;
        
       int i = 0, j = 0;

       for(i = 0; i < Arr.length; i++)
       {
           for(j = 0; j < Arr.length; j++)
           {
                if(Arr[i][j] == iNo)
                {
                    iCount++;
                }
           }
       }

       return iCount;
    }

}

class Program2
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter number of rows : ");
        int iRow = sobj.nextInt();

        System.out.println("Enter number of columns : ");
        int iCol = sobj.nextInt();

        System.out.println("Enter required number : ");
        int iNo = sobj.nextInt();

        Matrix mobj = new Matrix(iRow,iCol);

        mobj.Accept();

        mobj.Display();

        int iRet = 0;

        iRet = mobj.Frequency(iNo);

        System.out.println("Frequency of elements is : "+iRet);

    }
}